span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -28,16 +29,23 @@
   static final String SPANS_COUNTER = "hypertrace.reported.spans";
   private static final ConcurrentMap<String, Counter> statusToSpansCounter =
       new ConcurrentHashMap<>();
+
+  private final GrpcChannelRegistry grpcChannelRegistry;
+
   private TenantIdHandler tenantIdHandler;
   private SpanDropManager spanDropManager;
   private TagsFilter tagsFilter;
 
-  public JaegerSpanPreProcessor() {
-    // empty constructor
+  public JaegerSpanPreProcessor(GrpcChannelRegistry grpcChannelRegistry) {
```

Comment:
This isn't new'd programmatically or anything that forced the no-arg constructor is it? And if it's not, why did we have both in the first place?

Reply:
I think second one is just for testing..